README-developers.md

```diff
@@ -10,3 +10,108 @@
+production image to the production servers.
+
+### Development Work in a Personal Fork Repository
+
+#### Cloning the Primary Repository
+
+You can clone the primary repository normally. This document shows example
+commands for this repository.
+
+`git clone https://github.com/online-bridge-hackathon/common-docs.git`
+
+#### Creating a Personal Fork
+
+Go to the component page in the GitHub website. You can follow the
+[Forking Projects](https://guides.github.com/activities/forking/) guide.
+
+#### Adding the Personal Fork as a remote
+
+Git support multiple remote repositories for a local working tree. `git remote`
+command can be used to manage remote repositories. In most cases,
+`git remote add` is the only command required.
+
+`git remote add personal https://github.com/<username>/common-docs.git`
+
+More details:
+* [Adding a remote](https://docs.github.com/en/github/using-git/adding-a-remote)
+* [Working with Remotes](https://git-scm.com/book/en/Git-Basics-Working-with-Remotes)
+
+#### Creating a Development branch
+
+You can create a new local branch from the latest code in the `origin/master`
+remote branch. Basing a new development branch on the latest code makes merging
+the changes easier.
+
+```
+git fetch origin
+git checkout origin/master -b my_cool_new_feature
+```
+
+Commands create a new branch called `my_cool_new_feature` from the
+`origin/master` remote branch.
+
+#### Make All Commits Working (git bisect)
+
+All commits should pass unit tests. Bugs will happen, but the test suite reduces
+chance for a broken commit.
+
+Why all commits should be pass unit tests?
+
+Bisect functionality can help locate a regression bug. Using `git bisect` command can quickly find
+the first commit causing the bug. Bisect command uses a binary search like
+algorithm to find the first broken commit. The algorithm is modified to work
+well with a complex merge tree from many merges to the `master` branch.
+
+#### Making a Pull Request
+
+It is time to make a pull request when the new cool feature is ready and
+committed to the local branch. Pull request requires a new branch in the
+`personal` remote. `git push` can upload a local branch to a remote repository.
+
+`git push --set-upstream personal`
+
+The command pushes the `my_cool_new_feature` to `personal/my_cool_new_feature`.
+The `--set-upstream` argument updates the remote tracking information for the
+local branch. The update branch state makes subsequent `git push` select
+`personal/my_cool_new_feature` without any arguments.
+
+The GitHub adds a helpful link to the push command message. The link opens the pull request creation web page.
+
+#### Making changes based on review comments
+
+If the review process spots any issues,  then `git rebase -i` can be a useful
+command. The `-i` argument tells rebase to use the interactive mode.
+Interactive mode allows the author to modify new commits based on review
+comments.
+
+`git rebase -i origin/master`
```

Comment:
Please add a reference to help for this command.

Reply:
There was a reference link at the end of section. Should it be easier to notice?

I added a text "See more details about modifying history from"

Reply:
"git rebase -i " was very useful recently :)

Reply:
I see the reference now!